motionplan/motionPlanner_test.go

```diff
@@ -250,3 +258,199 @@ func testPlanner(t *testing.T, plannerFunc seededPlannerConstructor, config plan
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, spatialmath.PoseAlmostCoincidentEps(solvedPose.(*frame.PoseInFrame).Pose(), goal1, 0.01), test.ShouldBeTrue)
+}
+
+func TestMultiArmSolve(t *testing.T) {
+	fs := makeTestFS(t)
+	positions := frame.StartPositions(fs)
+	// Solve such that the ur5 and xArm are pointing at each other, 60mm from gripper to camera
+	goal2 := spatialmath.NewPoseFromOrientation(r3.Vector{Z: 60}, &spatialmath.OrientationVectorDegrees{OZ: -1})
+	newPos, err := PlanMotion(
+		context.Background(),
+		logger.Sugar(),
+		frame.NewPoseInFrame("urCamera", goal2),
+		fs.Frame("xArmVgripper"),
+		positions,
+		fs,
+		nil,
+		map[string]interface{}{"max_ik_solutions": 100, "timeout": 150.0},
+	)
+	test.That(t, err, test.ShouldBeNil)
+
+	// Both frames should wind up at the goal relative to one another
+	solvedPose, err := fs.Transform(newPos[len(newPos)-1], frame.NewPoseInFrame("xArmVgripper", spatialmath.NewZeroPose()), "urCamera")
+	test.That(t, err, test.ShouldBeNil)
+	solvedPose2, err := fs.Transform(newPos[len(newPos)-1], frame.NewPoseInFrame("urCamera", spatialmath.NewZeroPose()), "xArmVgripper")
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, spatialmath.PoseAlmostCoincidentEps(solvedPose.(*frame.PoseInFrame).Pose(), goal2, 0.1), test.ShouldBeTrue)
+	test.That(t, spatialmath.PoseAlmostCoincidentEps(solvedPose2.(*frame.PoseInFrame).Pose(), goal2, 0.1), test.ShouldBeTrue)
+}
+
+func TestSliceUniq(t *testing.T) {
+	fs := makeTestFS(t)
+	slice := []frame.Frame{}
+	slice = append(slice, fs.Frame("urCamera"))
+	slice = append(slice, fs.Frame("gantryOffset"))
+	slice = append(slice, fs.Frame("xArmVgripper"))
+	slice = append(slice, fs.Frame("urCamera"))
+	uniqd := uniqInPlaceSlice(slice)
+	test.That(t, len(uniqd), test.ShouldEqual, 3)
+}
+
+func TestSolverFrameGeometries(t *testing.T) {
+	fs := makeTestFS(t)
+	sFrames, err := fs.TracebackFrame(fs.Frame("xArmVgripper"))
+	test.That(t, err, test.ShouldBeNil)
+	sf, err := newSolverFrame(fs, sFrames, frame.World, frame.StartPositions(fs))
+	test.That(t, err, test.ShouldBeNil)
+
+	sfPlanner, err := newPlanManager(sf, fs, logger.Sugar(), 1)
+	test.That(t, err, test.ShouldBeNil)
+	position, err := sfPlanner.PlanSingleWaypoint(
+		context.Background(),
+		sf.sliceToMap(make([]frame.Input, len(sf.DoF()))),
+		spatialmath.NewPoseFromPoint(r3.Vector{300, 300, 100}),
+		nil,
+		nil,
+	)
+	test.That(t, err, test.ShouldBeNil)
+	gf, _ := sf.Geometries(position[len(position)-1])
+	test.That(t, gf, test.ShouldNotBeNil)
+	gripperCenter := gf.Geometries()["xArmVgripper"].Pose().Point()
+	test.That(t, spatialmath.R3VectorAlmostEqual(gripperCenter, r3.Vector{300, 300, 0}, 1e-2), test.ShouldBeTrue)
+}
+
+func TestMovementWithGripper(t *testing.T) {
+	// TODO(rb): move these tests to a separate repo eventually, as they take up too much time for general CI pipeline
+	t.Skip()
```

Comment:
With perf improvements, is it still desirable to skip this test?

Reply:
With perf improvements, this test takes 0.3 seconds on my machine, so I think we can un-skip

Reply:
One of the things that test does, is deliberately give CBiRRT an impossible path to solve; so it necessarily takes  a really long time, because it *must* time out.